clients/go/admin/auth_interceptor.go

```diff
@@ -102,3 +156,18 @@ func NewAuthInterceptor(cfg *Config, tokenCache cache.TokenCache, credentialsFut
 		return err
 	}
 }
+
+func NewProxyAuthInterceptor(cfg *Config, proxyCredentialsFuture *PerRPCCredentialsFuture) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+
+		err := invoker(ctx, method, req, reply, cc, opts...)
+		if err != nil {
+			newErr := MaterializeProxyAuthCredentials(ctx, cfg, proxyCredentialsFuture)
```

Comment:
The external command is called a single time per call of `flytectl`. After the call of the external command, the credentials future is initialized and can be reused for further grpc calls.